### PR TITLE
Fix: download retry handler

### PR DESCRIPTION
**Pull request by bigcat88**
As proposed in this issue https://github.com/nextcloud/integration_onedrive/issues/47#issuecomment-2291208701 by @baywet we are adding retry handler with help of `KiotaMiddleware`

### CODE REVIEW

**Comment by bigcat88** [comment]
Then only changed part is this:

```php
private function getKiotaHandlerStack(): HandlerStack {
    $handlerStack = HandlerStack::create();
    $handlerStack->push(KiotaMiddleware::retry(), RetryHandler::HANDLER_NAME);
    return $handlerStack;
}
```


```php

$options = [
    'stream' => true,
    'timeout' => 0,
    'headers' => [
        'User-Agent' => 'Nextcloud Dropbox integration',
    ],
    'handler' => $this->getKiotaHandlerStack(),  # this one!
];
``` 			

in `fileRequest` and usual `request` (that handles calendar, contacts and directory listing)

File: lib/Service/OnedriveAPIService.php [hunk 1]
```diff
@@ -156,6 +135,7 @@ public function request(string $userId, string $endPoint, array $params = [], st
 					'Authorization' => 'bearer ' . $accessToken,
 					'User-Agent' => 'Nextcloud OneDrive integration'
```
Comment (baywet) [inline lib/Service/OnedriveAPIService.php:136]: The user agent is invalid [according to RFC9110](https://www.rfc-editor.org/rfc/rfc9110#name-user-agent)


> User-Agent = product *( RWS ( product / comment ) )
> product         = token ["/" product-version]
> product-version = token
> tchar = "!" / "#" / "$" / "%" / "&" / "'" / "*" / "+" / "-" / "." /
 "^" / "_" / "`" / "|" / "~" / DIGIT / ALPHA
> token = 1*tchar


(ALPHA being A-Z a-z)

a better value would be something like "nextcloud/nextCloudVersion nextcloud-onedrive/pluginversion"

Additionally, this should be set by a middleware handler as well to get it out of the control logic. Kiota has a [Telemetry Handler](https://github.com/microsoft/kiota-http-guzzle-php/blob/main/src/Middleware/TelemetryHandler.php) that can be used to do that.

Not saying this is blocking for this PR, we should probably create a separate issue for that.

**Review by baywet — APPROVED** [review]
Thanks for putting this together, I could never find the time.

File: lib/Service/OnedriveAPIService.php (hunk 1 above)
Comment (bigcat88) [reply lib/Service/OnedriveAPIService.php:136]: Thanks for letting me know about this, I created a separate issue so as not to forget about this problem.

And thanks again for the suggestion on how to fix the problem using KiotaMiddleware

**Pull request merged**